src/pymorize/cmorizer.py

```diff
@@ -288,6 +294,25 @@ def _check_is_subperiod(self):
         if errors:
             for err in errors:
                 logger.error(err)
+            raise errors[0]
+
+    def _check_units(self):
+        errors = []
+        for rule in self.rules:
+            for input_collection in rule.inputs:
+                try:
+                    filename = input_collection.files[0]
+                except IndexError:
+                    break
+                da = xr.open_dataarray(filename, use_cftime=True)
```

Comment:
You **should not** open the data array to check anything!

Reply:
okay. will rewrite that